File: src/node/playground/carrier_account_detail_grabber.js
const Easypost = require('@easypost/api');
const dotenv = require('dotenv');

dotenv.config({ path: '/Users/jhammond/git/easypost/easypost-tools/.env' });
const api = new Easypost(process.env.EASYPOST_PROD_API_KEY);

// Grab the details of a carrier account by the account name "eg: FedexAccount"
// USAGE: CARRIER_ACCOUNT_NAME=FedexAccount node carrier_account_detail_grabber.js

api.CarrierType.all().then((r) => {
    for (let i = 0; i < r.length; i++) {
        if (r[i].type == process.env.CARRIER_ACCOUNT_NAME) {
            console.log(JSON.stringify(r[i]));
        }
    }
});

File: src/python/playground/carrier_account_detail_grabber.py
import os

import easypost
from dotenv import load_dotenv

CARRIER_ACCOUNT_NAME = os.getenv('CARRIER_ACCOUNT_NAME')


def main():
    carrier = retrieve_carrier_by_name()
    print(carrier)


def retrieve_carrier_by_name():
    load_dotenv()
    easypost.api_key = os.getenv('EASYPOST_PROD_API_KEY')

    carrier_accounts = easypost.CarrierAccount.all()
    for carrier in carrier_accounts:
        if carrier.type == CARRIER_ACCOUNT_NAME:
            return carrier
        else:
            return f'{CARRIER_ACCOUNT_NAME} not found!'


if __name__ == '__main__':
    main()
